datadog/resource_datadog_dashboard.go

```diff
@@ -4722,6 +4722,257 @@ func buildTerraformTimeseriesDefinition(datadogDefinition datadogV1.TimeseriesWi
 	return terraformDefinition
 }
 
+func getFormulaSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"formula_expression": {
+					Type:        schema.TypeString,
+					Required:    true,
+					Description: "String expression built from queries, formulas and functions.",
+				},
+				"limit": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					Description: "Options for limiting results returned.",
+					MaxItems:    1,
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"count": {
+								Type:        schema.TypeInt,
+								Optional:    true,
+								Description: "Number of results to return",
+							},
+							"order": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Default:     "desc",
+								Description: "Direction of sort.",
+							},
+						},
+					},
+				},
+				"alias": {
+					Type:        schema.TypeString,
+					Optional:    true,
+					Description: "Expression alias.",
+				},
+			},
+		},
+	}
+}
+
+func getFormulaQuerySchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"metric_query": {
+					Type:        schema.TypeList,
+					Optional:    true,
+					MaxItems:    1,
+					Description: "A timeseries formula and functions metrics query.",
+					Elem: &schema.Resource{
+						Schema: map[string]*schema.Schema{
+							"data_source": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Default:     "metrics",
+								Description: "Data source for metrics queries.",
+							},
+							"query": {
+								Type:        schema.TypeString,
+								Required:    true,
+								Description: "Metrics query definition.",
+							},
+							"aggregator": {
+								Type:        schema.TypeString,
+								Optional:    true,
+								Description: "The aggregation methods available for metrics queries.",
```

Comment:
This is an enum, you should use the enum validator.

Reply:
Done in [commit 2f914c9](https://github.com/DataDog/terraform-provider-datadog/pull/892/commits/2f914c9da8e833c9a46c5c24bd654aa47ba44918)